[Luo Siyi] iP

nicole-luo-exe opened the pull request:
### Ligma

> "What I have is not a dream, because I will make it a reality." - Sasuke Uchiha ([source](https://www.youtube.com/watch?v=FI6hkOxz-yY&ab_channel=NarutoWorld))

Ligma helps you make your dreams a reality by helping you keep track of the things you need to do. It is

- text-based

- easy to learn

- ~~FAST~~ _SUPER_-FAST to use

All you need to do is:

1. download the jar file from [here](https://github.com/nicole-luo-exe/ip/releases/tag/A-Jar)
2. run it on your command line with `java -jar ip.jar`
3. add your tasks
4. watch as you become an _**ALPHA**_  :fire: :smiling_imp: :heart_eyes_cat:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here's a sneak peek of the `main` method:
```
public static void main(String[] args) {
        new Ligma("data/ligma.txt").run();
}
```

Commit:
Add Gradle support

Commit:
Implement skeletal version of Duke

Commit:
Add store and list functionality and customize bot

Commit:
Add ability to mark task as done

Commit:
Add 3 types of tasks and customize bot

Commit:
Add testing in text-ui-test

Commit:
Handle errors and update testing to match

Commit:
Add delete functionality and update testing

Commit:
Implement save to hard disk

Commit:
Implement LocalDate within Task

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch-Level-8 with master

nicole-luo-exe committed:
Incorporate OOP practices

nicole-luo-exe committed:
Organize into packages

nicole-luo-exe committed:
Organise into packages

nicole-luo-exe committed:
Fix directory

nicole-luo-exe committed:
Add JUnit tests for TaskList, Parser and Deadline

nicole-luo-exe committed:
Fix conflicts

nicole-luo-exe committed:
Add JavaDoc

nicole-luo-exe committed:
Alter to fit coding standards

nicole-luo-exe committed:
Add command for find and tweak code for better OOP

nicole-luo-exe committed:
Merge branch A-JavaDoc

nicole-luo-exe committed:
Merge branch-A-CodingStandard

nicole-luo-exe committed:
Merge branch-Level-9

nicole-luo-exe commented:
### Ligma

> "What I have is not a dream, because I will make it a reality." - Sasuke Uchiha ([source](https://www.youtube.com/watch?v=FI6hkOxz-yY&ab_channel=NarutoWorld))

Ligma helps you make your dreams a reality by helping you keep track of the things you need to do. It is

- text-based

- easy to learn

- ~~FAST~~_SUPER_-FAST to use

All you need to do is:

1. download the jar file from [here](https://github.com/nicole-luo-exe/ip/releases/tag/A-Jar)
2. run it on your command line with `java -jar ip.jar`
3. add your tasks
4. watch as you become an _**ALPHA**_  :fire: :smiling_imp: :heart_eyes_cat:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here's a sneak peek of the `main` method:
```
public static void main(String[] args) {
        new Ligma("data/ligma.txt").run();
}
```

nicole-luo-exe committed:
Merge add-gradle-support and implement checkstyle

Timothyoung97 commented on src/main/java/ligma/Parser.java:
```diff
@@ -0,0 +1,119 @@
+            if (splitInd == -1) {
+                return handleBadCommands(action);
+            } else {
+                String description = input.substring(splitInd + 1);
+                int itemIndex = -1;
+                if (action.equals("done") || action.equals("delete")) {
+                    itemIndex = Character
+                            .getNumericValue(input.charAt(splitInd + 1)) - 1;
+                }
+                switch (action) {
+                case "todo":
+                    return new AddCommand(Task.TaskType.TODO,
+                            description);
+                case "event":
+                    return new AddCommand(Task.TaskType.EVENT,
+                            description);
+                case "deadline":
+                    return new AddCommand(Task.TaskType.DEADLINE,
+                            description);
+                case "done":
+                    return new DoneCommand(itemIndex);
+                case "delete":
+                    return new DeleteCommand(itemIndex);
+                case "find" :
+                    return new FindCommand(description.trim());
+                default:
+                    throw new NoSuchMethodException("Sorry, command does not exist.");
```
Maybe this part can be extracted out as another method to better readability?

nicole-luo-exe replied:
Good point! I'll see to making the necessary edits.

Timothyoung97 commented on src/main/java/ligma/Storage.java:
```diff
@@ -0,0 +1,105 @@
+ * Ligma program. The file f is where all data is written to
+ * and read from during the execution of the Ligma program.
+ */
+public class Storage {
```
I feel that this class is relatively easy to read and understand. I like how each method are relatively short and easy to interpret since the naming basically helps to explain the main purpose of the method. 👍

Timothyoung97 commented on src/main/java/ligma/task/Event.java:
```diff
@@ -0,0 +1,71 @@
+            return new Event(desc.substring(0, i - 1),
+                    time,
+                    isDone);
+        } else {
+            throw new InputMismatchException();
```
Good detection of error here. Help to catch formatting errors early 👍

Timothyoung97 commented on src/main/java/ligma/Ligma.java:
```diff
@@ -0,0 +1,58 @@
+import java.util.Scanner;
+
+import ligma.command.Command;
+/**
+ * This class represents a Ligma program that keeps track of your to-do list.
+ * Storage object corresponds to the persistent storage
+ * for the Ligma program executed and TaskList object corresponds
+ * to the existing tasks during execution.
+ */
```
Overall, I like how all classes are well documented such that they are very to read and understand without spending much time.

nicole-luo-exe replied:
Thanks for taking the time to review my repo! Much appreciated.

Timothyoung97 reviewed (commented):
LGTM 👍

1. Classes are relatively easy to read and understand.
2. All methods are mostly kept within 30 lines which helps in readability.
3. Naming of variables, classes and methods are pretty much self-explanatory.
4. SLAP observed 

Keep up the good work 💪

KengXIII commented on src/main/java/ligma/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int noOfTasks;
```
The o and 0 in 'noOfTasks' is a little difficult to read at first glance, maybe you could consider spelling it out in full

nicole-luo-exe replied:
I'll take note of that. Thanks!

erwinqxy replied:
Maybe you can consider `numOfTasks`?

nicole-luo-exe replied:
Sounds good, I'll change it to that. Thank you!

KengXIII reviewed (commented):
The code looks great, I don't spot many coding convention violations. The structure looks amazing and well thought out, it really shows the effort you put into designing Ligma. Good job!

nicole-luo-exe committed:
Add basic GUI

nicole-luo-exe committed:
Add basic GUI

nicole-luo-exe committed:
Merge branch 'branch-Level-10'

erwinqxy commented on src/main/java/ligma/Ligma.java:
```diff
@@ -0,0 +1,42 @@
+        this.tasks = new TaskList(storage.load());
+    }
+
+    public String getResponse(String input) {
```
Hi! I think you missed out on Javadoc for this method!

nicole-luo-exe replied:
Oh right, I have yet to add in all my javadocs. Apologies!

erwinqxy commented on src/main/java/ligma/Main.java:
```diff
@@ -0,0 +1,34 @@
+    private Ligma ligma;
+
+    @Override
+    public void start(Stage stage) {
```
Same here, you might have missed out on Javadoc for this method!

erwinqxy commented on src/main/java/ligma/Storage.java:
```diff
@@ -0,0 +1,105 @@
+     * @param task          task to be updated
+     * @throws IOException  if there was an error in writing to file
+     */
+    public void markDone(Task task) throws IOException {
```
I like how you placed a line break to improve readability

erwinqxy commented on src/main/java/ligma/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+     */
+    public String[] getStringArr() {
+        return tasks.stream()
+                .map(Task::toString)
```
Good practice of line break to improve readability! :smile:

erwinqxy commented on src/main/java/ligma/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+     * @return array of tasks that contains the target string
+     */
+    public Task[] find(String target) {
+        return tasks.stream().filter(task -> task.match(target)).toArray(Task[]::new);
```
Do you want to consider line breaks here to improve readability?

nicole-luo-exe replied:
Good point!

erwinqxy reviewed (commented):
The code is very readable and clean. I like how you aligned the words in your javadocs, very neat! Overall, not many coding violations and very good flow! Keep up the good work! :smile:

nicole-luo-exe committed:
Add asserts to Todo to ward against improper file format

nicole-luo-exe committed:
Tweak GUI

nicole-luo-exe committed:
Refactor code

Code has some coding quality issues re length of methods.

Methods that exceed 30 LoC can be hard to read.

Extracting methods can make it easier for others to understand
the code better.

Let's
* extract methods in a logical way to keep the nody under 30 LoC

nicole-luo-exe committed:
Merge pull request #1 from nicole-luo-exe/branch-A-Assertions

Add asserts to Todo to ward against improper file format

nicole-luo-exe committed:
Fix conflicts

nicole-luo-exe committed:
Merge pull request #2 from nicole-luo-exe/branch-A-CodeQuality

Merge branch-A-CodeQuality

nicole-luo-exe committed:
Add recurring tasks

Currently, only one-off tasks can be added.

This gives limited functionality to the user.

Creating a Recurring task would be good.

Let's
* add a task that shows when its next occurence is
* make the task update the next occurence based on current time

nicole-luo-exe committed:
Add help command

nicole-luo-exe committed:
Add Ui.png for user guide

nicole-luo-exe committed:
Add other screenshots of Ligma

nicole-luo-exe committed:
Add screenshot for find command

nicole-luo-exe committed:
Update README.md

nicole-luo-exe committed:
Set theme jekyll-theme-midnight

nicole-luo-exe committed:
Set theme jekyll-theme-minimal

nicole-luo-exe committed:
Set theme jekyll-theme-cayman

nicole-luo-exe committed:
Fix line break showing in Github pages

nicole-luo-exe committed:
Set theme jekyll-theme-cayman

nicole-luo-exe committed:
Link release to README.md

nicole-luo-exe committed:
Correct errors in javadoc

nicole-luo-exe committed:
Merge branch 'master' of https://github.com/nicole-luo-exe/ip

nicole-luo-exe committed:
Delete ligmaData directory
